### ISSUE TITLE
hooks for vterm-copy-mode enter and exit?

### ISSUE THREAD

**Issue by samcramer** [akermu/emacs-libvterm#579]
I'd like to make the window background color different when vterm is in vterm-copy-mode so I can easily see when I'm in copy mode.  I use [minions.el ](https://github.com/tarsius/minions), so the mode line doesn't indicate this in my case.  I think this could be done with vterm-copy-mode hooks.  It would be great if you could provide either those hooks or a more direct way to do this.

Awesome package, BTW!

**Comment by samcramer** [comment]
Never mind -- I think this can be done by setting the binding of C-c C-t to a wrapper for vterm-copy-mode.

**Comment by Sbozzolo** [comment]
This might be useful for other reasons, I will keep it open. If anyone reads this message and wants this feature, please use a reaction button to express your interest.

**Comment by psionic-k** [comment]
It's just cleaner than advice, but advice or decorating commands is fine for commands until they get used by other packages

**Comment by falood** [comment]
the enter copy mode hook could be done with `(add-hook 'vterm-copy-mode-hook xxxx)`, and not able to know whether the copy mode is done.
I need this feature to modify the mode-line color to let myself know whether I'm in copy mode

**Comment by ggcr** [comment]
Agree, we need a hook for when leaving copy-mode, just like we have it when we get into copy mode

Something like `(add-hook 'vterm-copy-mode-hook xxxx)` and `(add-hook 'vterm-copy-mode-exit-hook xxxx)`

**Comment by sandinmyjoints** [comment]
Agreed!

**Comment by njsmith** [comment]
`vterm-copy-mode-hook` already runs on both entry and exit -- this is apparently just how `define-minor-mode` works. Your hook can check `if vterm-copy-mode ...` at the beginning to figure out which way it's going.
